### ISSUE TITLE
Bug: Fix idea for entity incompatibility with any kind of spell.

### ISSUE THREAD

**Issue by SirAron111** [Electroblob77/Wizardry#209]
Minecraft version: 1.12.2
Wizardry version: 4.2.1
Environment: Singleplayer

Issue details: Bug: Fix idea for entity incompatibility with any kind of spell.

I had numerous crashes and glitches with spells trying to affect entities that don’t like to be affected by anything. 

For example midnight:rift (a not targetable entity that exists in 2 dimensions at the same time acting like a living portal) or dynamictrees:falling_tree (the entity that spawns looking like the tree that the player cut down doing a falling animation and then dropping its loot) there are many more such entities and I think adding in compatibility for all such entities will provide difficult.

So I thought about why you don’t add a blacklist where players/modpack makers can add special entities that simply get ignored not affected by any kind of spell. So entities like this could be added to this list and spells will simply phase trough them as if they don’t exist! Instead of trying to for example force field with the trinket that makes it do damage tries to reflect damage to a falling tree and it crashes. Or other things like that.

I mean I have no idea how to code but I think this would be a possibilities to fix a few of the special interaction bugs. (would also enable the possibility to make certain mobs immune to all magic if one so decides to for example add certain bosses to be unaffected by magic (I would do so for the twilight lich for example so it doesn’t get ported around by black hole or the like))

PS: I will on the next weekend take some time to check all spells and scrolls against all types of things to find more crashes and bugs like with the meteor. Sadly didn’t have the time until then.

GZ
SirAron

Link to crash report (if applicable): Would be to many different ones to post

**Comment by AzureDoom** [comment]
You need to update to 4.2.1

**Comment by SirAron111** [comment]
a ups i did have 4.2.1 was my bad did a typeo there XD because lazy me copy pastad it from my old post so i dont have to type it Xd

**Comment by Electroblob77** [comment]
Okay, I think there could be some mileage in this but I need to pinpoint exactly which situations cause problems and why. Normally, anything that isn't considered 'living' by the game (extends `EntityLivingBase`) will be ignored anyway... _surely_ trees and rifts aren't living?! [Edit: just checked, they aren't]

Do you have an example crash report, or are there already existing issues for this?

(Sorry, but it's a bit strange to post a solution without fully telling me the problem... 😕)

**Comment by SirAron111** [comment]
This one was mostly just an idea for a blacklist feature for entities that should be ignored by spells (in which I mean all spells ignore this entity projectiles phase trough it as If its not there, force field lets the mob or whatever pass trough as if it wasn’t there etc) This blacklist could be used for further customization (maybe say you want to make a boss mob immune to all magic) or to trouble shoot special entities added from mosd that cause problems now and maybe in the future. Until the potential problem with the potential entity can be properly fixed!

The Crash mentioned above happened before I did my big testing spree on all spells and before I know about all the problems and where they connected up. I had a dyn tree fall on my force field and it crashed me (later it turned out that this crash was just another one related to the revenge target thingy). I don’t have the crash report anymore. Sry

I made a more precise post on all my crash findings in a big collection where I pinpointed to some rough core problems with the majority of the crashes happening right now.
